dlio_benchmark/data_generator/npz_generator.py

```diff
@@ -51,8 +54,19 @@ def generate(self):
             out_path_spec = self.storage.get_uri(self._file_list[i])
             progress(i+1, self.total_files_to_generate, "Generating NPZ Data")
             prev_out_spec = out_path_spec
-            if self.compression != Compression.ZIP:
-                np.savez(out_path_spec, x=records, y=record_labels)
+
```

Comment:
Create a new class which inherits NPZGenerator and switch on generator_factory based on storage type.

Reply:
@hariharan-devarajan, currently, GeneratorFactory only receives the format type (NPZ). Should I pass the storage type during the initialization, so that it can select either NPZS3Generator or NPZGenerator class?